app/stores/DocumentsStore.ts

```diff
@@ -359,6 +361,19 @@ export default class DocumentsStore extends Store<Document> {
   fetchOwned = (options?: PaginationParams): Promise<Document[]> =>
     this.fetchNamedPage("list", options);
 
+  @action
+  fetchAll = async (): Promise<Document[]> => {
```

Comment:
I would just move this to the BaseStore, and pages to fetch should be based on the response not what happens to be cached in store already – the pagination data can be accessed with the symbol, eg `response[PAGINATION_SYMBOL].total`

Reply:
Sorry, shouldn't it be based on the cached items? How on `response`? It doesn't even exist in the scope of the method. What am I missing... :thinking:

Reply:
Are you suggesting we make an initial `fetchPage` request to get the value of `total` and then subsequently iterate and fetch pages _based on_ total?

Reply:
That or looping until there's no results left